src/platforms/evdev/libinput_device.cpp

```diff
@@ -694,3 +724,21 @@ void mie::LibInputDevice::apply_settings(mi::TouchscreenSettings const& settings
 {
     touchscreen = settings;
 }
+
+void mie::LibInputDevice::associate_to_id(MirInputDeviceId id)
+{
+    try_stop_observing_leds();
+    led_observer = std::make_shared<LibInputDeviceLedObserver>(this);
+    led_observer_registrar->register_interest(led_observer, id);
+    device_id = id;
+}
```

Comment:
This code looks like initialization, but is not in a constructor. Is there a good reason it is here? (And not, say, in `mi::DefaultDevice`?)

Reply:
The reason it is here is to not break the input ABI, but perhaps I'm not seeing something ;)